`bindgen!`, `async_trait` seem to be lost

```rust
wasmtime::component::bindgen!({
    world: "http",
    async: true,
});

impl HttpImports for HttpState {
    async fn request(&mut self, url: String) -> anyhow::Result<String> {
        println!("url is: {}", url);
        Ok(url.clone())
    }
}

async fn eval_with_imported() -> anyhow::Result<String> {

    let mut config = Config::new();
    config.async_support(true);
    config.wasm_component_model(true);
    let engine = Engine::new(&config)?;
    let component =
        wasmtime::component::Component::from_file(&engine, "../pasm-wit/pasm-wit-component.wasm")?;

    let mut linker = wasmtime::component::Linker::new(&engine);
    Http::add_to_linker(&mut linker, |state: &mut HttpState| state)?;

    let mut store = Store::new(&engine, HttpState {});
    let (bindings, _) = Http::instantiate_async(&mut store, &component, &linker).await?;

    let res = bindings.call_mainfn(&mut store).await?;

    Ok(res)
}

```

fails with error:

```
error[E0706]: functions in traits cannot be declared `async`
  --> src/main.rs:9:1
   |
9  | / wasmtime::component::bindgen!({
10 | |     world: "http",
11 | |     async: true,
12 | | });
   | |__^ `async` because of this
   |
   = note: `async` trait functions are not currently supported
   = note: consider using the `async-trait` crate: https://crates.io/crates/async-trait
   = note: see issue #91611 <https://github.com/rust-lang/rust/issues/91611> for more information
   = help: add `#![feature(async_fn_in_trait)]` to the crate attributes to enable
   = note: this error originates in the macro `wasmtime::component::bindgen` (in Nightly builds, run with -Z macro-backtrace for more info)

```

according to [doc](https://github.com/bytecodealliance/wasmtime/blob/main/crates/wasmtime/src/component/mod.rs#L266), `HttpImports` is supposed to have `async_trait` but I don't see it in `cargo expand`: 

```rust
use wasmtime::*;
pub struct Http {
    mainfn: wasmtime::component::Func,
}
pub trait HttpImports {
    async fn request(&mut self, url: String) -> anyhow::Result<String>;
}
```

also posted on [zulip](https://bytecodealliance.zulipchat.com/#narrow/stream/217126-wasmtime/topic/async_trait.2C.20bindgen!.20and.20wasmtime.3A.3Acomponent)

--- a/crates/component-macro/tests/codegen/direct-import.wit
+++ b/crates/component-macro/tests/codegen/direct-import.wit
@@ -1,0 +1,3 @@
+default world foo {
+  import foo: func()
+}

--- a/crates/wit-bindgen/src/lib.rs
+++ b/crates/wit-bindgen/src/lib.rs
@@ -364,6 +364,9 @@ impl Wasmtime {
         }
 
         let world_camel = resolve.worlds[world].name.to_upper_camel_case();
+        if self.opts.async_ {
+            uwriteln!(self.src, "#[wasmtime::component::__internal::async_trait]")
+        }
         uwriteln!(self.src, "pub trait {world_camel}Imports {{");
         for sig in functions {
             self.src.push_str(sig);

